activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -186,6 +176,20 @@ def fetch(key, *extras)
       super(convert_key(key), *extras)
     end
 
+    # Same as <tt>Hash#dig</tt> where the key passed as argument can be
+    # either a string or a symbol:
+    #
+    #   counters = ActiveSupport::HashWithIndifferentAccess.new
+    #   counters[:foo] = { bar: 1 }
+    #
+    #   counters.dig('foo', 'bar')     # => 1
+    #   counters.dig(:foo, :bar)       # => 1
+    #   counters.dig(:zoo)             # => nil
+    def dig(*args)
```

Comment:
Dig is only present on Ruby 2.3 and we still support Ruby 2.2. So we need to define it only conditionally.